src/components/Editor/Runners/HtmlRunner/HtmlRunner.jsx

```diff
@@ -118,20 +124,22 @@ function HtmlRunner() {
     window.addEventListener("message", (event) => {
       if (typeof event.data?.msg === "string") {
         if (event.data?.msg === "ERROR: External link") {
-          dispatch(setError("externalLink"));
-          showModal();
+          handleExternalLinkError();
         } else if (event.data?.msg === "Allowed external link") {
-          setExternalLink(event.data.payload.linkTo);
-          dispatch(triggerCodeRun());
+          handleAllowedExternalLink(event.data.payload.linkTo);
         } else {
-          setExternalLink(null);
-          setPreviewFile(`${event.data.payload.linkTo}.html`);
-          dispatch(triggerCodeRun());
+          handleRegularExternalLink(event.data.payload.linkTo);
```

Comment:
🔧 you need to define `setPreviewFile`

Reply:
it's at the line 77, should I move inside the new file?

Reply:
the function expects 2 arguments passed to is:
```
const handleRegularExternalLink = (linkTo, setPreviewFile)
```
but it's only getting the one at the min, so needs to be:
```suggestion
          handleRegularExternalLink(event.data.payload.linkTo, setPreviewFile);
```